src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs

```diff
@@ -0,0 +1,203 @@
+                return notApplicable;
+            }
+
+            if (portableExecutable.IsResourceOnly)
+            {
+                reasonForNotAnalyzing = MetadataConditions.ImageIsResourceOnlyBinary;
+                return notApplicable;
+            }
+
+            if (portableExecutable.IsNativeUniversalWindowsPlatform)
+            {
+                reasonForNotAnalyzing = MetadataConditions.ImageIsNativeUniversalWindowsPlatformBinary;
+                return notApplicable;
+            }
+
+            reasonForNotAnalyzing = null;
+            return AnalysisApplicability.ApplicableToSpecifiedTarget;
+        }
+
+        public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext context)
+        {
+            PEBinary target = context.PEBinary();
+            Pdb pdb = target.Pdb;
+
+            var notCompiledUsingCurrentDialectsModules = new List<AnalyzeResult>();
+
+            foreach (DisposableEnumerableView<Symbol> omView in pdb.CreateObjectModuleIterator())
+            {
+                Symbol om = omView.Value;
+                ObjectModuleDetails details = om.GetObjectModuleDetails();
+
+                if (details.Name != details.Library)
+                {
+                    continue;
+                }
+
+                string[] cVersion;
+                string[] cVersionExcluded = null;
+                if (details.WellKnownCompiler == WellKnownCompilers.MicrosoftC)
+                {
+                    cVersion = new string[] { "std:c" };
+                    cVersionExcluded = new string[] { "std:c++" };
+                }
+                else if (details.WellKnownCompiler == WellKnownCompilers.MicrosoftCxx)
+                {
+                    cVersion = new string[] { "std:c++" };
+                }
+                else
+                {
+                    continue;
+                }
+
+                if (!details.HasDebugInfo)
+                {
+                    continue;
+                }
+
+                string cVersionNumberString = string.Empty;
+                details.GetOptionValue(cVersion, OrderOfPrecedence.FirstWins, ref cVersionNumberString, cVersionExcluded);
+                AnalyzeResultType resultType = AnalyzeResultType.NotCurrent;
+
+                if (string.IsNullOrWhiteSpace(cVersionNumberString))
+                {
+                    resultType = AnalyzeResultType.NotSet;
+                }
+                else
+                {
+                    int cVersionNumber;
+                    if (cVersionNumberString.Equals("latest", StringComparison.OrdinalIgnoreCase))
+                    {
+                        resultType = AnalyzeResultType.Latest;
+                    }
+                    else if (int.TryParse(cVersionNumberString, out cVersionNumber))
+                    {
+                        resultType = cVersionNumber >= 17 ? AnalyzeResultType.Current : AnalyzeResultType.NotCurrent;
```

Comment:
>17 [](http://example.com/codeflow?start=55&length=2)

Ah! I just noticed that e.g. C99 with number 99 will be greater then 17. I will re-write this when back on this task.